snow/consensus/snowman/bootstrapper/bootstrapper.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bootstrapper
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+// Bootstrapper implements the protocol used to determine the initial set of
+// accepted blocks to sync to.
+//
+// The bootstrapping protocol starts by fetching the last accepted block from an
+// initial subset of peers. In order for the protocol to find a recently
+// accepted block, there must be at least one correct node in this subset of
+// peers. If there is not a correct node in the subset of peers, the node will
+// not accept an incorrect block. However, the node may be unable to find an
+// acceptable block.
+//
+// Once the last accepted blocks have been fetched from the subset of peers, the
+// set of blocks are sent to all peers. Each peer is expected to filter the
+// provided blocks and report which of them they consider accepted. If a
+// majority of the peers report that a block is accepted, then the node will
+// consider that block to be accepted by the network. This assumes that a
+// majority of the network is correct. If a majority of the network is
+// malicious, the node may accept an incorrect block.
+type Bootstrapper interface {
```

Comment:
Chatted offline about seeing if we can dedup this interface and just have two separate implementations for GetAcceptedFrontiers and GetAccepted